src/utils/ProviderUtils.ts

```diff
@@ -574,22 +574,48 @@ export async function getProvider(chainId: number, logger?: winston.Logger, useC
 }
 
 export function getNodeUrlList(chainId: number): string[] {
-  const retryConfigKey = `NODE_URLS_${chainId}`;
-  if (process.env[retryConfigKey]) {
-    const nodeUrls = JSON.parse(process.env[retryConfigKey]) || [];
-    if (nodeUrls?.length === 0) {
-      throw new Error(`Provided ${retryConfigKey}, but parsing it as json did not result in an array of urls.`);
+  const resolveUrls = (): string[] => {
+    const providers = process.env[`RPC_PROVIDERS_${chainId}`] ?? process.env["RPC_PROVIDERS"];
+    if (providers === undefined) {
+      throw new Error(`No RPC providers defined for chainId ${chainId}`);
     }
+    const nodeUrls = providers.split(",").map((provider) => {
+      const envVar = `RPC_PROVIDER_${provider}_${chainId}`;
+      const url = process.env[envVar];
+      if (url === undefined) {
+        throw new Error(`Missing RPC provider URL (${envVar})`);
```

Comment:
```suggestion
        throw new Error(`Missing RPC provider URL for chain ${chainId} (${envVar})`);
```

Something like this to make it more clear?